occurrences/tests/test_api.py

```diff
@@ -877,3 +877,73 @@ def test_unenrol_occurrence(snapshot, user_api_client):
     executed = user_api_client.execute(UNENROL_OCCURRENCE_MUTATION, variables=variables)
     snapshot.assert_match(executed)
     assert occurrence.study_groups.count() == 0
+
+
+def test_occurrences_filter_by_date(api_client, snapshot):
+    OccurrenceFactory(
+        start_time=datetime(2020, 1, 1, 0, 0, 0, tzinfo=timezone.now().tzinfo),
+    )
+    OccurrenceFactory(
+        start_time=datetime(2020, 1, 2, 0, 0, 0, tzinfo=timezone.now().tzinfo),
+    )
+    variables = {"date": "2020-01-02"}
+    executed = api_client.execute(OCCURRENCES_QUERY, variables=variables)
+
+    assert len(executed["data"]["occurrences"]["edges"]) == 1
+    OccurrenceFactory(
+        start_time=datetime(2020, 1, 2, 0, 0, 0, tzinfo=timezone.now().tzinfo),
+    )
+    executed = api_client.execute(OCCURRENCES_QUERY, variables=variables)
+    assert len(executed["data"]["occurrences"]["edges"]) == 2
+    snapshot.assert_match(executed)
+
+
+def test_occurrences_filter_by_time(api_client, snapshot):
+    for i in range(10, 12):
+        OccurrenceFactory(
+            start_time=datetime(2020, 1, 1, i, 0, 0, tzinfo=timezone.now().tzinfo),
+        )
+        OccurrenceFactory(
+            start_time=datetime(2020, 1, 2, i + 1, 0, 0, tzinfo=timezone.now().tzinfo),
+        )
+    OccurrenceFactory(
+        start_time=datetime(2020, 1, 1, 13, 0, 0, tzinfo=timezone.now().tzinfo),
+    )
+    variables_1 = {"time": "12:00:00"}
+    variables_2 = {"time": "14:00:00+02:00"}
+    variables_3 = {"time": "11:00:00+00:00"}
+
+    executed = api_client.execute(OCCURRENCES_QUERY, variables=variables_1)
+    assert len(executed["data"]["occurrences"]["edges"]) == 1
+    executed = api_client.execute(OCCURRENCES_QUERY, variables=variables_2)
+    assert len(executed["data"]["occurrences"]["edges"]) == 1
```

Comment:
I think most of the time with tests like this it is also good to somehow verify that the returned results are exactly the objects you are expecting, by comparing IDs or something. I've been bitten by that a few times in the past, where a test passes because the right amount of results are returned by coincidence, but actually the results have been wrong 🙂

Reply:
Agree! I think I missed the snapshot assertion here, there are other test cases which snapshots were used